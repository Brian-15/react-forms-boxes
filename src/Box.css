.Box {
    display: flex;
    flex-direction: row;
    margin-top: 3em;
}

.Box-body {
    margin-right: 1em;
}

.Box-remove-btn {
    margin: auto;
    width: 2em;
    height: 2em;
}