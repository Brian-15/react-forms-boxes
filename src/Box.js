import React from 'react';
import './Box.css';

const Box = ({ removeBox, id, ...style }) => {
    return (
        <div className='Box' id={id}>
            <div className='Box-body' style={style} data-testid="box" />
            <button className='Box-remove-btn' onClick={removeBox}>X</button>
        </div>
    );
};

export default Box;