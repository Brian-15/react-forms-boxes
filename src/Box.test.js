import { render } from "@testing-library/react";
import Box from './Box.js';

test('smoke test: renders box', () => {
    render(
        <Box
            backgroundColor='blue'
            width='200px'
            height='200px'
        />);
});

test('matches snapshot', () => {
    const { asFragment } = render(
        <Box
            backgroundColor='blue'
            width='200px'
            height='200px'
        />
    );
    expect(asFragment()).toMatchSnapshot();
});