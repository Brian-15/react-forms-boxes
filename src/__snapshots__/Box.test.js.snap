// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-body"
      style="background-color: blue; width: 200px; height: 200px;"
    />
    <button
      class="Box-remove-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
